write-fonts/src/graph/splitting.rs

```diff
@@ -630,12 +629,9 @@ fn size_of_value_record_children(
         .filter_map(|offset| (!offset.is_null()).then_some(*offset.offset()))
         .map(|_| {
             let obj = offsets[*next_offset_idx].object;
-            if seen.insert(obj) {
-                *next_offset_idx += 1;
-                graph.objects[&obj].bytes.len()
-            } else {
-                0
-            }
+            let seen = seen.insert(obj);
+            *next_offset_idx += 1;
+            seen.then(|| graph.objects[&obj].bytes.len()).unwrap_or(0)
```

Comment:
I dont' immediately see why you need to bind seen, wouldn't this be fine as:

```rust
            let obj = offsets[*next_offset_idx].object;
            *next_offset_idx += 1;
            seen.insert(obj)
                    .then(|| graph.objects[&obj].bytes.len())
                    .unwrap_or(0)
```

Hm. Actually your `map` always takes an offset and increments next which sounds a lot like iteration so just zip?

```rust
    subtables
        .iter()
        .filter_map(|offset| (!offset.is_null()).then_some(*offset.offset()))
       .zip(offsets.iter()).
        .map(|(offset, _)| {
            seen.insert(offset.object)
                    .then(|| graph.objects[&offset.object].bytes.len()).unwrap_or(0)
        })
        .sum()
```